ansible_runner/display_callback/events.py

```diff
@@ -28,6 +28,9 @@
 import threading
 import uuid
 
+# Ansible
+from ansible.parsing.ajson import AnsibleJSONEncoder
```

Comment:
This is heading into some dangerous territory as Runner doesn't import Ansible directly right now. How portable is this? Do we *need* to do this?

Reply:
This bug throws errors in the callback plugin, I would say it's a must-fix.

The alternative is to copy over @ryanpetrello's

https://github.com/ansible/awx/blob/devel/awx/main/queue.py#L23

We could switch AWX to import runner's version of the encoder, eventually. That could move us in the direction of making ansible-runner the de-facto python library for Ansible variable encoder / decoders. I don't necessarily think that's a bad idea either.

Keep in mind, AWX already has _other_ encoders and decoders for unsafe variables. We may have yet un-discovered bugs for other types.

Reply:
I think I'd definitely like to pull in *that* old encoder if it gets us there. I'd like to keep from importing Ansible *at all* if there is literally any other way around an issue like this.

Reply:
here? https://github.com/ansible/ansible-runner/blob/master/ansible_runner/utils.py

Reply:
No, wait, I have even more questions. If I put this in utils, it still wouldn't be a simple import. In cases where these files import from other parts of ansible_runner, it doesn't import from the ansible_runner module.

https://github.com/ansible/ansible-runner/blob/master/ansible_runner/display_callback/module.py#L33

There's a lot of baggage with how the imports are done. I don't necessarily feel more confident importing from another file (wherever that may be) in ansible_runner. So I guess the option on the table is to just define the class right in that file.

Reply:
If we think we might expand on this in the future, which seems likely, then I feel like we should possibly create a `ansible_runner/encoders.py`